src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringInterpolation.cs

```diff
@@ -35,6 +36,108 @@ private BoundExpression RewriteInterpolatedStringConversion(BoundConversion conv
+            var usesBoolReturn = node.InterpolationData.BuilderFormatCalls[0].Method.ReturnType.SpecialType == SpecialType.System_Boolean;
+            var tryFormatCalls = ArrayBuilder<BoundExpression>.GetInstance(node.InterpolationData.BuilderFormatCalls.Length);
+            for (int i = 0; i < node.Parts.Length; i++)
+            {
+                var formatInfo = node.InterpolationData.BuilderFormatCalls[i];
+                var part = node.Parts[i];
+                Debug.Assert(usesBoolReturn == (formatInfo.Method.ReturnType.SpecialType == SpecialType.System_Boolean));
+                Debug.Assert(formatInfo.Arguments[0] is BoundInterpolatedStringElementPlaceholder { Index: var index } && index == i);
+                Debug.Assert(part is BoundStringInsert or BoundLiteral);
+
+                // Visit the part, and replace the part placeholder in the TryFormat... call with the replaced part.
+                var visitedPart = VisitExpression(part is BoundStringInsert { Value: var value } ? value : part);
+                var formatCall = MakeCallWithNoExplicitArgument(formatInfo with
+                {
+                    Arguments = formatInfo.Arguments.SetItem(0, visitedPart)
+                }, part.Syntax, builderTemp, assertParametersAreOptional: false);
+                tryFormatCalls.Add(formatCall);
+                Debug.Assert(usesBoolReturn == (formatCall.Type!.SpecialType == SpecialType.System_Boolean));
+            }
+
+            var builderPatternStatements = ArrayBuilder<BoundStatement>.GetInstance();
+            builderPatternStatements.Add(builderConstruction);
+            if (usesBoolReturn)
+            {
+                BoundExpression? tryFormatBinaryExpressions = null;
+
+                // builder.TryFormat... && builder.TryFormat... && ...
+                foreach (var tryFormatCall in tryFormatCalls)
+                {
+                    tryFormatBinaryExpressions = tryFormatBinaryExpressions is null
+                        ? tryFormatCall
+                        : _factory.LogicalAnd(tryFormatBinaryExpressions, tryFormatCall);
+                }
+
+                Debug.Assert(tryFormatBinaryExpressions is not null);
+
+                builderPatternStatements.Add(_factory.ExpressionStatement(tryFormatBinaryExpressions));
+            }
+            else
+            {
+                // builder.TryFormat...; builder.TryFormat...; ...
+                foreach (var tryFormatCall in tryFormatCalls)
+                {
+                    builderPatternStatements.Add(_factory.ExpressionStatement(tryFormatCall));
+                }
+            }
+
+            tryFormatCalls.Free();
+
+            // PROTOTYPE(interp-string): This will need to be a lot more complicated to support method parameter scenarios
+            // resultTemp = builder.ToString();
+
+            builderPatternStatements.Add(_factory.Assignment(resultTemp, _factory.InstanceCall(builderTemp, "ToString")));
```

Comment:
>InstanceCall [](start = 82, length = 12)

It looks like this is the first usage of the helper. Either we should find an alternative, or a very thorough testing should be performed to ensure we get expected behavior in various cases (method is missing, inaccessible, etc.)